uast/nodes/nodesproto/nodesproto.go

```diff
@@ -433,6 +483,33 @@ func (g *graphReader) asTree() (nodes.Node, error) {
 	seen := make(map[uint64]bool, len(g.nodes))
 	return g.asNode(g.root, seen)
 }
+
+func (m *Node) Kind() nodes.Kind {
+	if m.Value != nil {
+		v, _ := asValue(m)
+		return nodes.KindOf(v)
+	}
+	if len(m.Keys) != 0 || m.IsObject {
+		return nodes.KindObject
+	}
+	return nodes.KindArray
+}
+
+func asValue(n *Node) (nodes.Value, error) {
```

Comment:
Is it possible to make this method public?

Reply:
Why do you need it? The `RawNode` has a `Value` method that stores an output of this function.

Reply:
true. My bad.